spec.md

```diff
@@ -8,121 +8,162 @@ title: Semantic Specification
     [Span C········································]
-         [Span E···········]   [Span F········]
-                                [Span G······]
-                                [Span H··]
-                                        [Span I·]
+         [Span E·······]        [Span F··] [Span G··] [Span H··]
 ~~~
 
 ## Traces
 
-A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a tree of Spans. (See the ASCII diagrams above)
+A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system. A Trace can be thought of as a directed acyclic graph (DAG) of Spans.
 
 ## Spans
 
-A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model parent-child and causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely names the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask within a larger computation).
+A **Span** represents a logical unit of work in the system that has a start time and a duration. Spans may be nested and ordered to model causal relationships. Each span has an **operation name**, a presumably human-readable string which concisely represents the work done by the span (e.g., an RPC method name, a function name, or the name of a subtask or stage within a larger computation).
+
+<span id="references"></span>
+
+### Causal Span References
+
+A Span may reference zero or more Spans that are causally related. OpenTracing presently defines two types of references:
+
+**Parent references:** A "parent" Span may create a new "child" Span. The expectation is that the parent Span has not finished by the time the new Span starts. Though it is often the case that the parent Span cannot complete successfully until the child Span does as well, this is not a semantic requirement for parent references.
+
+A timing diagram for a parent that's blocked on a child Span:
+
+~~~
+    [-Parent Span---------]
+         [-Child Span----]
+~~~
+
+A timing diagram for a parent that's not blocked on a child Span:
+
+~~~
+    [-Parent Span-]
+              [-Child Span-]
+~~~
+
+**Sequence predecessor references:** Some Spans cannot start until a "sequence predecessor" Span finishes. For instance, chained sequence predecessor references neatly describe pipeline stages separated by queues.
+
+A sequence predecessor Span is part of the same logical trace as the descendant Span that refers to it: i.e., the descendant Span is formally caused by the work of its sequence predecessor.
+
+A timing diagram for a SequencePredecessorRef that must finish before the descendant Span can start:
+
+~~~
+    [-SequencePredecessorRef Span-]  [-New Span-]
+~~~
+
 
```

Comment:
passing note.. there's a fairly important new concept kited in this PR, which one can familiarize with by reading the go implementation. Maybe the semantics of this can be in a separate topic or PR in this repo? Looking at the title, this PR is basically a rename.. looking at the comments, you see a powerful new tool discussed (sequential predecessor etc)